### PR TITLE
VEN-1443 | Add table export functionality to customers and application views

### PR DESCRIPTION
**Pull request by Sartonon**
## Description :sparkles:

Adds table export functionality.

Looping the results might not be needed but lets see how the API evolves.

### CODE REVIEW

File: src/common/utils/__tests__/useTableExport.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { renderHook } from '@testing-library/react-hooks';
+import * as Apollo from '@apollo/react-hooks';
+
+import { useTableExport } from '../useTableExport';
+import { exportClient } from '../../../app/exportClient';
+import * as downloadFile from '../downloadFile';
+// import { getBerthToken } from '../auth';
```
Comment (Jokinen) [inline src/common/utils/__tests__/useTableExport.test.ts:8]: FYI

File: src/features/applicationView/ApplicationViewContainer.tsx [hunk 2]
```diff
@@ -87,12 +88,14 @@ const ApplicationViewContainer = () => {
...
     });
-  const handleLinkCustomer = (customerId: string) =>
+  const handleLinkCustomer = (customerId: string) => {
     linkCustomer({
       variables: {
         input: { id, customerId },
       },
-    });
+      // eslint-disable-next-line @typescript-eslint/no-empty-function
+    }).catch(() => {});
```
Comment (Jokinen) [inline src/features/applicationView/ApplicationViewContainer.tsx:97]: Intentional?

**Review by Jokinen — APPROVED** [review]
The technical approach looks really really good here 👍

I'm mainly wondering about the UX. Seems like the label may be PoC ("Vie..").

But I'm also thinking about the "Toiminnot" dropdown 🤔 It's open by default which makes it a bit of a non issue, but it's not the most descriptive label. Of course, this is an admin UI so the admin can be taught so not a huge problem.

I'm also thinking whether there should be some kind of indicator that the excel is being downloaded/built?

**Comment by Sartonon** [comment]
> The technical approach looks really really good here 👍
> 
> I'm mainly wondering about the UX. Seems like the label may be PoC ("Vie..").
> 
> But I'm also thinking about the "Toiminnot" dropdown 🤔 It's open by default which makes it a bit of a non issue, but it's not the most descriptive label. Of course, this is an admin UI so the admin can be taught so not a huge problem.
> 
> I'm also thinking whether there should be some kind of indicator that the excel is being downloaded/built?

Now that I was testing all the export endpoints, I noticed missing loading indicator too. There should be some indicator for that. "Vie..." is from the designs but not the best label...

**Pull request merged**